src/document.c

```diff
@@ -875,12 +908,36 @@ const char *DocumentField_GetValueCStr(const DocumentField *df, size_t *len) {
     case FLD_VAR_T_CSTR:
       *len = df->strlen;
       return df->strval;
+    case FLD_VAR_T_ARRAY:
+      if (df->arrayLen > 0) {
+        // Return the first entry
+        *len = strlen(df->multiVal[0]);
+        return df->multiVal[0];
+      }
+      break;
     case FLD_VAR_T_NULL:
       break;
     case FLD_VAR_T_NUM:
-    case FLD_VAR_T_GEO:
-    case FLD_VAR_T_ARRAY:
+    case FLD_VAR_T_GEO:    
       RS_LOG_ASSERT(0, "invalid types");
   }
   return NULL;
 }
+
+const char *DocumentField_GetArrayValueCStr(const DocumentField *df, size_t *len, size_t index) {
+  if (df->unionType == FLD_VAR_T_ARRAY && index < df->arrayLen) {
+    *len = strlen(df->multiVal[index]);
+    return df->multiVal[index];
+  }
+  *len = 0;
+  return NULL;
+}
+
+size_t DocumentField_GetArrayValueCStrLen(const DocumentField *df) {
```

Comment:
So this returns the sum of all the strings len? If so maybe named it like TotalSize? Currently it looks like it return the len of a specific element.